### PR TITLE
RFC First pass at jamongo API

### PR DESCRIPTION
**Pull request by cjacoby**

### CODE REVIEW

File: jams/jamongo.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+This library provides an interface for interfacing with a JAMS Mongo
+instance.
+
+"""
+
+import pymongo
+
+
+def convert_annotation_list(annotations_array):
```
Comment (bmcfee) [inline jams/jamongo.py]: 8-space tabs! Copypasta

File: jams/jamongo.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        return self
+
+    def __exit__(self, type, value, traceback):
+        pass
+
+    @property
+    def db(self):
+        return self._db
+
+    @property
```
Comment (bmcfee) [inline jams/jamongo.py]: We have "annotations" and "audio" here... should there be somewhere that explicitly defines these collections?  And should these getters be attributes, or would it make more sense to provide a dict interface in this object where the key maps to a collection in `_db`?

File: jams/jamongo.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        jams_object : jams.JAMS
+        """
+        # Extract the JAMS metadata & insert it into the Audio collection
+        audio_id = self.insert_jams_metadata(jams_object.file_metadata)
+
+        # Extract the JAMS Annotations and prepare them for mongification.
+        annotation_list = convert_annotation_list(jams_object.annotations)
+
+        # insert them into the Annotations collection.
+        annotation_ids = self.insert_annotations(annotation_list, audio_id)
```
Comment (bmcfee) [inline jams/jamongo.py]: Maybe simpler to have a method that inserts one at a time, and then iterate over the annotation_list?

File: jams/jamongo.py (hunk 1 above)
Comment (cjacoby) [inline jams/jamongo.py]: drp. fixed it.

File: jams/jamongo.py (hunk 2 above)
Comment (cjacoby) [inline jams/jamongo.py]: re: "should there be somewhere that explicitly defines these collections":
+1 on that.

re: "And should these getters be attributes, or would it make more sense to provide a dict interface in this object where the key maps to a collection in _db"

Do you mean something like:
```self._collections = ["music", "annotations"]

def __getattr__(self, key):
    return self._collections[key]
```

???

File: jams/jamongo.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+    def db(self):
+        return self._db
+
+    def __getattr__(self, key):
+        """Get a collection object from the db, if it is an allowable
+        collection specified."""
+        if key in self._collections:
+            return self._db['key']
+        else:
+            return None
```
Comment (bmcfee) [inline jams/jamongo.py]: do we want a `return None` here, or throw a key error?

File: jams/jamongo.py (hunk 3 above)
Comment (bmcfee) [inline jams/jamongo.py]: !resolved

File: jams/jamongo.py (hunk 4 above)
Comment (cjacoby) [inline jams/jamongo.py]: ... yeah key error is better.

File: jams/jamongo.py (hunk 1 above)
Comment (bmcfee) [inline jams/jamongo.py]: !resolved

File: jams/jamongo.py (hunk 2 above)
Comment (bmcfee) [inline jams/jamongo.py]: !resolved

File: jams/jamongo.py [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+    Parameters
+    ----------
+    annotations_list : jams.AnnotationArray
+
+    audio_id : ObjectId
+        Pointer to an audio document.
+
+    Returns
+    -------
+    converted_annotations : list of Annotation
```
Comment (bmcfee) [inline jams/jamongo.py]: Just to be 100% clear, we for sure want deep-copied `jams.Annotation` objects returned here, and not simplified dicty json objects.  Right?

Comment (cjacoby) [inline jams/jamongo.py]: Yes. We'll just have to do a lil' old `[x.__json__ for x in annotations]` before we actually insert it, but I think this is what we want.

Comment (bmcfee) [inline jams/jamongo.py]: !resolved

File: jams/jamongo.py (hunk 4 above)
Comment (bmcfee) [inline jams/jamongo.py]: !resolved

**Pull request merged**